File: gang/week8/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = nums.length / 2;   // 선택 마리
        HashSet<Integer> result = new HashSet<>();  // 중복 제거

        for(int i=0; i< nums.length; i++) { // 중복된 값을 제외하고 값을 넣음
            result.add(nums[i]);
        }

        return Math.min(result.size(), answer);     // 둘 중에 작은 수가 가장 많은 수 선택
    }
}

File: jeonghyeon/8week/폰켓몬.js
function solution(nums) { // 폰켓몬 종류 개수를 구하는 문제(가질 수 있는 최대의 종류 수)
    const choosenum = nums.length / 2; // 가져갈 폰켓몬 수 총 4마리 => 2마리
    const eachkind = [...new Set(nums)]; // 중복 제거한 폰켓몬 종류 수 [3,1,2]
    // 3개의 종류가 있으니 2개 뽑아가면 됨
    // 가져갈 폰켓몬 수 < 중복 제거 폰켓몬 종류 수 => 가져갈 폰켓몬 수
    if(choosenum < eachkind.length) {
        return choosenum;
    } else {
        return eachkind.length;
    }
}